src/main/java/edu/umich/its/cpm/EmailFormatter.java

```diff
@@ -130,6 +278,22 @@ public String getBody() {
         ArrayList<String> headers = (ArrayList<String>) messageMap.get("headers");
         return headers;
     }
+    public String getItemId(){
+        ArrayList<String> headers = getHeaders();
+        String date=null;
+        String subject=null;
+        String itemId;
+        for (String header: headers) {
+            if (header.startsWith(DATE_WITH_COLON)) {
+                 date = header.split(DATE_WITH_COLON)[1].trim();
+            }
+            if(header.startsWith(SUBJECT_WITH_COLON)){
+                 subject = header.split(SUBJECT_WITH_COLON)[1].trim();
+            }
+        }
+        itemId=date+" "+subject;
```

Comment:
This unique identifier goes in the reporting, for End user perspective this is how they can identify message Subject, added date incase 2 messages have same subject name. With respective logs when the email message migration start msg Id(Primary key of the table record) is being displayed , i think developers can trace back very easily. I doubt that BSA will ever look at logs.....